### PR TITLE
JBPM-6135:Add Bulk actions on jobs list

### PR DESCRIPTION
**Pull request by nmirasch**

### CODE REVIEW

**Comment by nmirasch** [comment]
@cristianonicolai  @bsremac Could you take a look please? Thanks!

File: jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListViewImpl.java [hunk 1]
```diff
@@ -145,4 +173,70 @@ public void initSelectionModel(ListTable<RequestSummary> extendedPagedTable) {
                         presenter.getViewProcessActionCondition(),
                         true));
     }
+
+    private void initBulkActions(final ExtendedPagedTable<RequestSummary> extendedPagedTable) {
```
Comment (cristianonicolai) [inline jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListViewImpl.java]: This could be moved into _AbstractMultiGridView_ so each view just have to return a list of bulk actions to be added right? :)

File: jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListViewImpl.java [hunk 2]
```diff
@@ -118,21 +140,27 @@ public void initSelectionModel(ListTable<RequestSummary> extendedPagedTable) {
                 new ConditionalAction<>(
                         constants.Cancel(),
                         job -> {
-                            if (Window.confirm(constants.CancelJob())) {
```
Comment (cristianonicolai) [inline jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListViewImpl.java]: thanks for removing this! :)

**Comment by nmirasch** [comment]
@cristianonicolai added the change you suggested :)

File: jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/resources/org/jbpm/workbench/es/client/i18n/Constants.properties [hunk 3]
```diff
@@ -105,4 +100,21 @@ UnacknowledgedErrors=Unackowledged Execution Errors
 AcknowledgedErrors=Acknowledged Execution Errors
 ViewProcessInstance=View process
 ViewTask=View task
-RequestRequeued=Request {0} put back into execution queue
\ No newline at end of file
+RequestRequeued=Request {0} put back into execution queue
+Cancel_Job=Cancel job
+Cancel_Job_Sure=Are you sure about to cancel this Job?
```
Comment (bsremac) [inline jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/resources/org/jbpm/workbench/es/client/i18n/Constants.properties]: Why was this changed? It sounded better to me how it was i.e. ```Are you sure you want to cancel this Job?```
The same comment about the new bulk messages.

Also, one minor thing regarding the capitalization. When we are asking for confirmation about one job, ```J``` in ```Job``` is a capital letter, and when there's a bulk of them ```j``` is a small letter in ```jobs```

File: jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/test/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListPresenterTest.java [hunk 4]
```diff
@@ -514,4 +516,74 @@ public void testOnBasicFilterRemoveEvent() {
...
+
+        ArrayList<RequestSummary> requestSummaries= new ArrayList<>();
+        requestSummaries.add(createRequestSummary(jobId_1,
+                                                  key,
+                                                  deploymentId,
+                                                  RequestStatus.QUEUED));
+        requestSummaries.add(createRequestSummary(jobId_2,
+                                                  key,
+                                                  deploymentId,
+                                                  RequestStatus.QUEUED));
```
Comment (bsremac) [inline jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/test/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListPresenterTest.java]: Maybe changing this one to ```RequestStatus.RUNNING```, since we already have one queued?

**Comment by nmirasch** [comment]
@bsremac done!

File: jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/resources/org/jbpm/workbench/es/client/i18n/Constants.properties [hunk 5]
```diff
@@ -105,4 +100,21 @@ UnacknowledgedErrors=Unackowledged Execution Errors
...
+RequestRequeued=Request {0} put back into execution queue
+Cancel_Job=Cancel job
+Cancel_Job_Sure=Are you sure you want to cancel this Job?
+Cancel_Job_Confirmation=Cancel job confirmation
+Requeue_Job=Requeue job
+Requeue_Job_Sure=Are you sure you want to requeue this Job?
+Requeue_Job_Confirmation=Requeue job confirmation
+Bulk_Cancel_jobs=Bulk Cancel
+Cancel_Jobs=Cancel jobs
+Cancel_Jobs_Sure=Are you sure you want to cancel this Jobs?
```
Comment (bsremac) [inline jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/resources/org/jbpm/workbench/es/client/i18n/Constants.properties]: I feel bad to comment again, but could you please just change in the case of bulk actions the message from ```this Jobs``` to ```these Jobs```? :)

Comment (nmirasch) [reply jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/resources/org/jbpm/workbench/es/client/i18n/Constants.properties]: @bsremac ups! my fault. Thanks for the comment!

**Comment by nmirasch** [comment]
@bsremac changed!

**Comment by bsremac** [comment]
@nmirasch Muchas gracias! :)

File: jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListViewImpl.java [hunk 6]
```diff
@@ -145,4 +176,46 @@ public void initSelectionModel(ListTable<RequestSummary> extendedPagedTable) {
...
+                                                                    getCancelCommand(extendedPagedTable))
+        );
+
+        return bulkAbortNavLink;
+    }
+
+    protected AnchorListItem getBulkRequeue(final ExtendedPagedTable<RequestSummary> extendedPagedTable) {
+        final AnchorListItem bulkAbortNavLink = GWT.create(AnchorListItem.class);
+        bulkAbortNavLink.setText(constants.Bulk_Requeue_jobs());
+        bulkAbortNavLink.setIcon(IconType.BAN);
```
Comment (cristianonicolai) [inline jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListViewImpl.java]: Perhaps use _pficon-restart_?

File: jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/errorlist/ExecutionErrorListViewImpl.java [hunk 7]
```diff
@@ -93,26 +88,16 @@ public void initSelectionModel(final ListTable<ExecutionErrorSummary> extendedPa
...
-                                                          Style.Unit.PX);
-        bulkButton.setEnabled(false);
-        bulkActions.add(bulkButton);
-        final DropDownMenu bulkDropDown = GWT.create(DropDownMenu.class);
-        bulkDropDown.addStyleName(Styles.DROPDOWN_MENU + "-right");
-        bulkDropDown.getElement().getStyle().setMarginRight(5,
-                                                            Style.Unit.PX);
-        bulkDropDown.add(bulkAckNavLink);
-        bulkActions.add(bulkDropDown);
         bulkAckNavLink.setIcon(IconType.BAN);
```
Comment (cristianonicolai) [inline jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/errorlist/ExecutionErrorListViewImpl.java]: Perhaps use _pficon-orders_

**Comment by nmirasch** [comment]
@cristianonicolai, @bsremac I added this ones. 
jobbulkactionsicons

**Comment by bsremac** [comment]
@nmirasch Me gusta :)

**Comment by cristianonicolai** [comment]
@nmirasch this one is probably nice to add to N&N
